File: Scala/src/main/scala/adapter/TurkeyAdapter.scala
package adapter

import strategy.Duck

class TurkeyAdapter(turkey: Turkey) extends Duck {
  override val flyBehavior: () => Unit = turkey.fly
  override val quackBehavior: () => Unit = turkey.gobble

  override def display(): Unit = println(turkey.getClass.getSimpleName + "-Adapter")
}

File: Kotlin/src/main/kotlin/chapter7_adapter/TurkeyAdapter.kt
package chapter7_adapter

// TurkeyAdapter는 Turkey를 상속받아, Duck을 구현하고 있다
class TurkeyAdapter: Turkey(), Duck {
    override fun quack() {
        gobble()
    }
}

// WildTurkeyAdapter는 WildTurkey를 상속받아, Duck을 구현하고 있다
class WildTurkeyAdapter: WildTurkey(), Duck {
    override fun quack() {
        gobble()
    }
}
